searxng/searxng
# [simple] make checkboxes accessible

## What does this PR do?

* The checkboxes can get the focus
* The checkboxes are clickable on 
* In addition, engine tab in the preferences:
  * the checkboxes get an ARIA label
  * the engine name is label : the user can click on a engine name to enable / disable the engine.
* trim down the number of HTML element to show a checkbox.
* remove some spaces here and there in `preferences.html`
* the `/preferences` page size is divided by two (140ko instead of 299ko).

![image](https://user-images.githubusercontent.com/1594191/155855530-7c158bd4-dd6e-4145-ae37-91759fcdc979.png)

![image](https://user-images.githubusercontent.com/1594191/155855504-2a7e9b7a-4d07-425f-a3ca-b8f69f1b7879.png)

---

Do note that checkboxes in the engines tab are displayed reversed (checked is displayed unchecked) because of:
* https://github.com/searxng/searxng/blob/3408d061aab9abc6168fec9bbc6deab71b236dac/searx/templates/simple/preferences.html#L313
* code in `preferences.py`

The CSS  class `reversed-checkbox` allows to reverse the checkbox meaning.
* Implementation : https://github.com/searxng/searxng/blob/31b6118e3f8d8553ea4267b6760fc3659c8b7a0b/searx/static/themes/simple/src/less/toolkit.less#L427-L453
* Usage : https://github.com/searxng/searxng/blob/31b6118e3f8d8553ea4267b6760fc3659c8b7a0b/searx/templates/simple/preferences.html#L101

## Why is this change important?

Improve the web accessibility

## How to test this PR locally?

* check the `/preferences` page (light and dark versions):
   * change the engine selections, save, go back to the preferences, check 
   * click on a engine name
   * use tab, shift+tab

## Author's checklist



## Related issues

Related to https://github.com/searxng/searxng/issues/818
